src/RA_Dlg_MemBookmark.cpp

```diff
@@ -539,31 +536,38 @@ void Dlg_MemBookmark::SetupColumns(HWND hList)
 
     //	Remove all data.
     ListView_DeleteAllItems(hList);
+    {
+        auto lplvColumn{ std::make_unique<LV_COLUMN>() };
+        auto idx{ 0 };
+        for (auto& sColTitle : ra::COLUMN_TITLE)
+        {
+            lplvColumn->mask       = LVCF_TEXT | LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT;
+            lplvColumn->fmt        = LVCFMT_CENTER | LVCFMT_FIXED_WIDTH;
+            lplvColumn->cx         = ra::COLUMN_WIDTH.at(idx);
 
-    LV_COLUMN col;
-    ZeroMemory(&col, sizeof(col));
+            // NB: The column title has to cached before-hand or we won't see it. -SBS
+            ra::tstring tszText{ sColTitle };
+            lplvColumn->pszText    = tszText.data();
+            lplvColumn->cchTextMax = 255;
+            lplvColumn->iSubItem   = idx;
 
-    for (size_t i = 0; i < NumColumns; ++i)
-    {
-        col.mask = LVCF_TEXT | LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT;
-        col.cx = COLUMN_WIDTH[i];
-        ra::tstring colTitle = NativeStr(COLUMN_TITLE[i]).c_str();
-        col.pszText = const_cast<LPTSTR>(colTitle.c_str());
-        col.cchTextMax = 255;
-        col.iSubItem = i;
-
-        col.fmt = LVCFMT_CENTER | LVCFMT_FIXED_WIDTH;
-        if (i == NumColumns - 1)
-            col.fmt |= LVCFMT_FILL;
-
-        ListView_InsertColumn(hList, i, (LPARAM)&col);
-    }
+            if (idx == (ra::COLUMN_TITLE.size() - 1))
+                lplvColumn->fmt |= LVCFMT_FILL;
 
+            ListView_InsertColumn(hList, idx, lplvColumn.get());
+            idx++;
+        }
+    }
     m_nNumOccupiedRows = 0;
 
-    BOOL bSuccess = ListView_SetExtendedListViewStyle(hList, LVS_EX_FULLROWSELECT | LVS_EX_GRIDLINES | LVS_EX_DOUBLEBUFFER);
-    bSuccess = ListView_EnableGroupView(hList, FALSE);
-
+    auto bSuccess = ListView_SetExtendedListViewStyle(hList, LVS_EX_FULLROWSELECT | LVS_EX_GRIDLINES | LVS_EX_DOUBLEBUFFER);
+#if _WIN32
+    bSuccess = ra::to_unsigned(ListView_EnableGroupView(hList, FALSE));
+#elif _WIN64
+    bSuccess = static_cast<DWORD>(ra::to_unsigned(ListView_EnableGroupView(hList, FALSE)));
+#else
+#error Unknown platform, Windows is currently the only supported platform!
+#endif // _WIN32
```

Comment:
`bSuccess` isn't even used. Why go through all this trouble for it?

Reply:
I like being thorough, it seems you'd prefer to just check the return value instead. That'll be less trouble.

Reply:
i was just commenting that the return value wasn't being used, so you didn't need seven lines of code to make sure it was set correctly.

Reply:
I got you. It did seem like a waste.